dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,721 @@
+                linkMap.put("name", link.getTitle());
+                linkMap.put("title", link.getName());
+                linkMap.put("description", link.getFriendlyName());
+                linkMap.put("extension", "link");
+                linkMap.put("hasLiveVersion", APILocator.getVersionableAPI().hasLiveVersion(link));
+                linkMap.put("statusIcons", UtilHTML.getStatusIcons(link));
+                linkMap.put("hasTitleImage", "");
+                linkMap.put("__icon__", "linkIcon");
+                returnList.add(linkMap);
+            }
+        }
+    } // includeLinks.
+
+    private void includeFolders(final BrowserQuery browserQuery,
+                                final List<Map<String, Object>> returnList,
+                                final Role[] roles,
+                                final Folder parent) throws DotDataException, DotSecurityException {
+
+        if (parent != null) {
+
+            List<Folder> folders = Collections.emptyList();
+            try {
+
+                folders = folderAPI.findSubFolders(parent, userAPI.getSystemUser(), false);
+            } catch (Exception e1) {
+
+                Logger.error(this, "Could not load folders : ", e1);
+            }
+
+            for (final Folder folder : folders) {
+
+                final List<Integer> permissions = permissionAPI.getPermissionIdsFromRoles(folder, roles, browserQuery.user);
+                if (permissions.contains(PERMISSION_READ)) {
+
+                    final Map<String, Object> folderMap = folder.getMap();
+                    folderMap.put("permissions", permissions);
+                    folderMap.put("parent", folder.getInode());
+                    folderMap.put("mimeType", "");
+                    folderMap.put("name", folder.getName());
+                    folderMap.put("title", folder.getName());
+                    folderMap.put("description", folder.getTitle());
+                    folderMap.put("extension", "folder");
+                    folderMap.put("hasTitleImage", StringPool.BLANK);
+                    folderMap.put("__icon__", IconType.FOLDER.iconName());
+                    returnList.add(folderMap);
+                }
+            }
+        }
+    } // includeFolders.
+
+
+    private Map<String,Object> htmlPageMap(final HTMLPageAsset page,
+                                           final User user,
+                                           final boolean showArchived,
+                                           final long languageId) throws DotDataException, DotStateException, DotSecurityException {
+
+	        final Map<String, Object> pageMap = new HashMap<>(page.getMap());
+
+            pageMap.put("mimeType",     "application/dotpage");
+            pageMap.put("name",         page.getPageUrl());
+            pageMap.put("description",  page.getFriendlyName());
+            pageMap.put("extension",    "page");
+            pageMap.put("isContentlet", true);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "isContentlet" appears 4 times in this file; the first occurrence is on line 602](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)